nntrainer/graph/network_graph.cpp

```diff
@@ -411,8 +411,11 @@ void NetworkGraph::setMaxExecutionOrder(bool skip_optimize) {
         std::max(max_exec_order, std::get<1>(ln->getExecutionOrder()));
 #endif
     } else {
+      /// FIXME below code need explanation
+      //   max_exec_order =
+      //   std::max(max_exec_order, std::get<0>(ln->getExecutionOrder()));
       max_exec_order =
-        std::max(max_exec_order, std::get<0>(ln->getExecutionOrder()));
+        std::max(max_exec_order, std::get<2>(ln->getExecutionOrder()));
```

Comment:
I wonder why it is 2? it checks 
```
ln->needsCalcDerivative() || ln->needsCalcGradient()
``` 
at line 405. Should n't it be 0?

Reply:
Yes, this should be 0. This is being corrected at #1833 :)